### PR TITLE
feat: terramate experimental cloud login

### PR DESCRIPTION
**Pull request by i4ki**
Implements the `terramate experimental cloud login`.
Adding `-v` provides detailed output.

- Add `user_terramate_dir` option to `.terramaterc` file (CLI config file)
- Login with Google Provider.
- Open the browser with the OAuth2 consent page.
- RedirectURI to 127.0.0.1:<randon port in range 40000-52023>
- Store token, refresh token and expiration in `$homeTerramateDir/credentials.tmrc.json`

Now it's also possible to customize where the checkpoint cache and signature are stored by setting the `homeTerramateDir`.

### CODE REVIEW

File: cmd/terramate/cli/cli.go [hunk 1]
```diff
@@ -112,6 +112,9 @@ type cliSpec struct {
 	DisableCheckpoint          bool `optional:"true" default:"false" help:"Disable checkpoint checks for updates"`
 	DisableCheckpointSignature bool `optional:"true" default:"false" help:"Disable checkpoint signature"`
 
+	Login struct {
+	} `cmd:"" help:"Login into Terramate Cloud" hidden:"true"`
+
```
Comment (mariux) [inline cmd/terramate/cli/cli.go:117]: let's go with `experimental cloud login` for now ;)

**Comment by soerenmartius** [comment]
Thanks @i4ki - I just tested this feature and it works great! Regarding UX: Can we automatically open the browser the login window instead of asking the user to manually copy the URL from inside the terminal? For an example see https://cloud.google.com/sdk/gcloud
Screenshot 2023-05-15 at 19 51 03

File: cmd/terramate/cli/cli.go (hunk 1 above)
Comment (i4ki) [reply cmd/terramate/cli/cli.go]: done!

**Comment by i4ki** [comment]
@soerenmartius I implemented the opening of the browser. Can you check if it works in the MacOS? (it's hard to create tests for that)

**Comment by i4ki** [comment]
nice, it fallback to print the URL if it fails to open in the browser.

**Review by soerenmartius — APPROVED** [review]
lgtm - awesome

**Review by soerenmartius — DISMISSED** [review]
lgtm - awesome

**Review by soerenmartius — APPROVED** [review]
lgtm

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**